utils/retrieve-state/src/lib.rs

```diff
@@ -171,38 +176,94 @@ pub async fn download_block_with_deploys(
     })
 }
 
+/// Download block files from the highest (provided) block hash to genesis.
```

Comment:
But `BlockIdentifier` isn't a hash...?

Reply:
BlockIdentifier can be a hash or a height. If it's a height we need to download the block and get the hash from there.

Reply:
```suggestion
/// Download block files from the block hash or height to genesis.
```

Maybe say this?